crates/core_simd/src/masks.rs

```diff
@@ -88,6 +88,10 @@ impl_element! { isize }
 /// The layout of this type is unspecified, and may change between platforms
 /// and/or Rust versions, and code should not assume that it is equivalent to
 /// `[T; LANES]`.
+///
+/// For a type with layout guaranteed equivalent to `[T; LANES]`, use
+/// `SIMD<T, LANES>`. For a type with layout guaranteed to use 1 bit per
+/// lane (padded up to full bytes), use `LANES::BitMask`.
```

Comment:
So, the exact number of bytes is an implementation detail (we've used `u64` before and we might choose it again... basically we want to retain the freedom to use an integer of a reasonable size, OR to use a byte array, depending on what strikes the best ergonomics/codegen balance). Right now, this seems like it might suggest it uses exactly the next byte in size. I am not sure what would be most clear to express our desired reservation while providing you the minimum of guarantees you want (or if you even consider that acceptable).